python/dgl/graphbolt/impl/feature_cache.py

```diff
@@ -0,0 +1,80 @@
+"""HugeCTR gpu_cache wrapper for graphbolt."""
+import torch
+
+__all__ = ["FeatureCache"]
+
+
+class FeatureCache(object):
+    r"""High level wrapper for the CPU feature cache.
+
+    Parameters
+    ----------
+    cache_shape : List[int]
+        The shape of the cache. cache_shape[0] gives us the capacity.
+    dtype : torch.dtype
+        The data type of the elements stored in the cache.
+    num_parts: int, optional
+        The number of cache partitions for parallelism. Default is 1.
+    policy: str, optional
+        The cache policy to be used. Default is "s3-fifo".
+    """
+
+    def __init__(self, cache_shape, dtype, num_parts=1, policy="s3-fifo"):
+        policies = ["s3-fifo"]
+        assert (
+            policy in policies
+        ), f"{policies} are the available caching policies."
+        assert num_parts >= 1
+        self._policy = (
+            torch.ops.graphbolt.s3_fifo_cache_policy(cache_shape[0])
+            if num_parts == 1
+            else torch.ops.graphbolt.partitioned_s3_fifo_cache_policy(
+                cache_shape[0], num_parts
```

Comment:
nit: maybe refactor the c++ code a little bit and explicitly define API for cache policy, such that we don't can just use `s3_fifo_cache_policy(cache_shape[0], num_parts)` and rely on num_parts to decide whether to use partitioned or not.

Reply:
Some other caching policies might have multiple arguments. Let's keep this simple solution for now and refactor the C++ code if and when the need arises. I feel this refactor belongs in another PR.

Reply:
Just changed the code so that the partitioned version directly calls and returns the result of the underlying policy if num_parts == 1.

Reply:
That way, we don't have to expose 2 classes to python. Thank you for the suggestion.